### ISSUE TITLE
new installation of v1.13.0 resulted in etcd container restarts

### ISSUE THREAD

**Issue by sbezverk** [kubernetes/kubeadm#1299]
## Is this a request for help?
Yes

it is a bug report

## Versions
 kubeadm version
kubeadm version: &version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.0", GitCommit:"ddf47ac13c1a9483ea035a79cd7c10005ff21a6d", GitTreeState:"clean", BuildDate:"2018-12-03T21:02:01Z", GoVersion:"go1.11.2", Compiler:"gc", Platform:"linux/amd64"}
**kubeadm version** (use `kubeadm version`):

**Environment**:
- **Kubernetes version** (use `kubectl version`):
kubectl version
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.0", GitCommit:"ddf47ac13c1a9483ea035a79cd7c10005ff21a6d", GitTreeState:"clean", BuildDate:"2018-12-03T21:04:45Z", GoVersion:"go1.11.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.0", GitCommit:"ddf47ac13c1a9483ea035a79cd7c10005ff21a6d", GitTreeState:"clean", BuildDate:"2018-12-03T20:56:12Z", GoVersion:"go1.11.2", Compiler:"gc", Platform:"linux/amd64"}
- **Cloud provider or hardware configuration**:
baremetal
- **OS** (e.g. from /etc/os-release):
cat /etc/os-release 
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"
- **Kernel** (e.g. `uname -a`):
Linux kube-5.sbezverk.cisco.com 4.16.0-1.el7.elrepo.x86_64 #1 SMP Sun Apr 1 20:13:35 EDT 2018 x86_64 x86_64 x86_64 GNU/Linux
- **Others**:

## What happened?

etcd container keeps restarting
## What you expected to happen?
etcd should be stable

## Anything else we need to know?

```
[root@kube-5 ~]# kubectl get configmap --all-namespaces
NAMESPACE     NAME                                 DATA   AGE
kube-public   cluster-info                         2      74m
kube-system   canal-config                         4      68m
kube-system   coredns                              1      74m
kube-system   extension-apiserver-authentication   6      74m
kube-system   kube-proxy                           2      74m
kube-system   kubeadm-config                       2      74m
kube-system   kubelet-config-1.13                  1      74m
[root@kube-5 ~]# kubectl get configmap --all-namespaces -o yaml
apiVersion: v1
items:
- apiVersion: v1
  data:
    jws-kubeconfig-pomtgp: eyJhbGciOiJIUzI1NiIsImtpZCI6InBvbXRncCJ9..cdM6U9DkmcPStjPM5KpBJVfFdX7LG_B1WKLOybkcvvs
    kubeconfig: |
      apiVersion: v1
      clusters:
      - cluster:
          certificate-authority-data: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUN5RENDQWJDZ0F3SUJBZ0lCQURBTkJna3Foa2lHOXcwQkFRc0ZBREFWTVJNd0VRWURWUVFERXdwcmRXSmwKY201bGRHVnpNQjRYRFRFNE1USXdOREU0TWpjMU1sb1hEVEk0TVRJd01URTRNamMxTWxvd0ZURVRNQkVHQTFVRQpBeE1LYTNWaVpYSnVaWFJsY3pDQ0FTSXdEUVlKS29aSWh2Y05BUUVCQlFBRGdnRVBBRENDQVFvQ2dnRUJBTXQ3Cml0MU1LKzFPRkZJZUE2WTdzcEZoa2xTZ3NoY3NtTEJLZkcxUG9ra2V6WnJZS0VjWVpmN1Rub0F5cjFIRGZwT2oKSFpieWpwZmpNeVNVanVZZkJmdUVPZzRnRm5FM01oZlpFT0FHcFJ5WU0yMitWYWFLY1VhblE1NXZ5Ums0NjV6bwo1dUlsWllDYmwzRmEwZU5yd0JKcS9XOWdCRGxhOHZlNFk2OGp3NGp2WGdsdjZZdFdaanpmNHlYSXNvYnRKbEJiCkVRRFpqZGU1NEpIT29BWUJ5U2JoVmdtSlQvZnQ3ZjF3ZE8wdzhjRUljaVdQZ1BsT0lDM1B3Z1NqQmtKNHJLRUgKNGZEZE5Eekd0OExUQTkrRHA2UkFhZTBPQlhaZThnT0YwTlFSa0x3a2ZoUFJET0c2M3YrOTJnRkQ0cGJ5SXprRAo5L3lyTU9xQThpd0tpeW1malBrQ0F3RUFBYU1qTUNFd0RnWURWUjBQQVFIL0JBUURBZ0trTUE4R0ExVWRFd0VCCi93UUZNQU1CQWY4d0RRWUpLb1pJaHZjTkFRRUxCUUFEZ2dFQkFKWEEwNWRKNlR2OFc2eDkyR1BmZ1NmakY1dkwKNlEwRzRkTVZzVDBZMHEzN3VQU3hGcms2VFQxelRkdGFsWkoyamZsd1BDZk9sQ3psQ2NmMEJPVC9PbVZjQ3JxRwpFU2ZZL2ZFajZxT09iQU44M3FRVjZjTElmWlFJRDFqNXk0THdTa1Uxb2NTM0JKZFZMSitYZ0piZkJCcExXbXg5Ck5ubUVLS0szeFFINHlnRzNqK0dYK0gwcWlzc2ZoaWJsb2tvQnBvaHVHSmdEYXVnVXF4bVVWNlZZeU9YenRlOHAKeFAxaVpwdWJtbFJiM3BzMVlDQ0J2QjZ1OEFZSFpiMnBONG9lSkdKR0RQVE9vVWVXd1pWM0l2RXYrSzhNaytNNQpiMEQ1MVRhbG1KNFdzdzEvdExvN3JQU1p3QzJqM3had29IdTlQL293WGptcTYwR2tpWTlubkl1T2tPUT0KLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=
          server: https://192.168.80.105:6443
        name: ""
      contexts: []
      current-context: ""
      kind: Config
      preferences: {}
      users: []
  kind: ConfigMap
  metadata:
    creationTimestamp: "2018-12-04T18:28:15Z"
    name: cluster-info
    namespace: kube-public
    resourceVersion: "361"
    selfLink: /api/v1/namespaces/kube-public/configmaps/cluster-info
    uid: 5d9f79a5-f7f2-11e8-9183-5254004c4249
- apiVersion: v1
  data:
    canal_iface: ""
    cni_network_config: |-
      {
        "name": "k8s-pod-network",
        "cniVersion": "0.3.0",
        "plugins": [
          {
            "type": "calico",
            "log_level": "info",
            "datastore_type": "kubernetes",
            "nodename": "__KUBERNETES_NODE_NAME__",
            "ipam": {
              "type": "host-local",
              "subnet": "usePodCidr"
            },
            "policy": {
                "type": "k8s"
            },
            "kubernetes": {
                "kubeconfig": "__KUBECONFIG_FILEPATH__"
            }
          },
          {
            "type": "portmap",
            "snat": true,
            "capabilities": {"portMappings": true}
          }
        ]
      }
    masquerade: "true"
    net-conf.json: |
      {
        "Network": "57.144.0.0/12",
        "Backend": {
          "Type": "vxlan"
        }
      }
  kind: ConfigMap
  metadata:
    creationTimestamp: "2018-12-04T18:34:54Z"
    name: canal-config
    namespace: kube-system
    resourceVersion: "950"
    selfLink: /api/v1/namespaces/kube-system/configmaps/canal-config
    uid: 4b570b9e-f7f3-11e8-9183-5254004c4249
- apiVersion: v1
  data:
    Corefile: |
      .:53 {
          errors
          health
          kubernetes sbezverk.cisco.com in-addr.arpa ip6.arpa {
             pods insecure
             upstream
             fallthrough in-addr.arpa ip6.arpa
          }
          prometheus :9153
          proxy . /etc/resolv.conf
          cache 30
          loop
          reload
          loadbalance
      }
  kind: ConfigMap
  metadata:
    creationTimestamp: "2018-12-04T18:28:15Z"
    name: coredns
    namespace: kube-system
    resourceVersion: "224"
    selfLink: /api/v1/namespaces/kube-system/configmaps/coredns
    uid: 5da0dc76-f7f2-11e8-9183-5254004c4249
- apiVersion: v1
  data:
    client-ca-file: |
      -----BEGIN CERTIFICATE-----
      MIICyDCCAbCgAwIBAgIBADANBgkqhkiG9w0BAQsFADAVMRMwEQYDVQQDEwprdWJl
      cm5ldGVzMB4XDTE4MTIwNDE4Mjc1MloXDTI4MTIwMTE4Mjc1MlowFTETMBEGA1UE
      AxMKa3ViZXJuZXRlczCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBAMt7
      it1MK+1OFFIeA6Y7spFhklSgshcsmLBKfG1PokkezZrYKEcYZf7TnoAyr1HDfpOj
      HZbyjpfjMySUjuYfBfuEOg4gFnE3MhfZEOAGpRyYM22+VaaKcUanQ55vyRk465zo
      5uIlZYCbl3Fa0eNrwBJq/W9gBDla8ve4Y68jw4jvXglv6YtWZjzf4yXIsobtJlBb
      EQDZjde54JHOoAYBySbhVgmJT/ft7f1wdO0w8cEIciWPgPlOIC3PwgSjBkJ4rKEH
      4fDdNDzGt8LTA9+Dp6RAae0OBXZe8gOF0NQRkLwkfhPRDOG63v+92gFD4pbyIzkD
      9/yrMOqA8iwKiymfjPkCAwEAAaMjMCEwDgYDVR0PAQH/BAQDAgKkMA8GA1UdEwEB
      /wQFMAMBAf8wDQYJKoZIhvcNAQELBQADggEBAJXA05dJ6Tv8W6x92GPfgSfjF5vL
      6Q0G4dMVsT0Y0q37uPSxFrk6TT1zTdtalZJ2jflwPCfOlCzlCcf0BOT/OmVcCrqG
      ESfY/fEj6qOObAN83qQV6cLIfZQID1j5y4LwSkU1ocS3BJdVLJ+XgJbfBBpLWmx9
      NnmEKKK3xQH4ygG3j+GX+H0qissfhiblokoBpohuGJgDaugUqxmUV6VYyOXzte8p
      xP1iZpubmlRb3ps1YCCBvB6u8AYHZb2pN4oeJGJGDPTOoUeWwZV3IvEv+K8Mk+M5
      b0D51TalmJ4Wsw1/tLo7rPSZwC2j3xZwoHu9P/owXjmq60GkiY9nnIuOkOQ=
      -----END CERTIFICATE-----
    requestheader-allowed-names: '["front-proxy-client"]'
    requestheader-client-ca-file: |
      -----BEGIN CERTIFICATE-----
      MIIC0DCCAbigAwIBAgIBADANBgkqhkiG9w0BAQsFADAZMRcwFQYDVQQDEw5mcm9u
      dC1wcm94eS1jYTAeFw0xODEyMDQxODI3NTNaFw0yODEyMDExODI3NTNaMBkxFzAV
      BgNVBAMTDmZyb250LXByb3h5LWNhMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIB
      CgKCAQEAqroBXOs5fLcbqz4+AcVGDkR3cRX22Lvka52tawQCC9bpRU8PTMk7qP5w
      fxGNg3FvZgJ/UF/O6GN7nRFaO/58+BFE6bekfkXvfIhFvJdM4o5fksNNRSCqy0AK
      m5zdILoznCB5S7Mn8dGSpTPjjoR1OWxUvyZKqx5Bno0YEtbQaBDqGFyGT+Jf7Td2
      GJ3NBKamB+54ZbS2p+N05Y14u/FKwKGMFcXl6uxOsjiMDXvxK4Q1C7FkgR9Di9wn
      CvNKthvcEJacr/rk4npPqzYrC+iik3ag+iotYM7IPFm9XuLaZZgk0JH0pi2AX1oM
      InVYw9G1kHxAy2CtHFpxi2p99x7DeQIDAQABoyMwITAOBgNVHQ8BAf8EBAMCAqQw
      DwYDVR0TAQH/BAUwAwEB/zANBgkqhkiG9w0BAQsFAAOCAQEARWOMh3DKu7NulrGS
      ny/d1Cc9AaFu+89EClHYR/2VN9Woa6tcHEIXteZInTcv5f2nQZafZ7FeOr/ypidM
      gJUJhji69ZuejoE/z27PF69UFfpHPbe1wOdor63Oh9AacrW+zjah6cBLzw8cEpRs
      q3meUlDY9CloGv6MXW/xmzFqoellCQMXCLZKTQHqmDnxuWjvWeHdezU4mE5fFC0u
      /ymGih+ekdxAjERzWGvee+jI3c/cxLeirjFp7TDI/rCqauZHyWjBphG4Inodjfgg
      XOzXxVbQInWwbPAuIxyB4aPNc+zmuIAQJTDWRXPQdhmC2SSVWw8zpqDiKDPKPt3i
      hE3XWg==
      -----END CERTIFICATE-----
    requestheader-extra-headers-prefix: '["X-Remote-Extra-"]'
    requestheader-group-headers: '["X-Remote-Group"]'
    requestheader-username-headers: '["X-Remote-User"]'
  kind: ConfigMap
  metadata:
    creationTimestamp: "2018-12-04T18:28:12Z"
    name: extension-apiserver-authentication
    namespace: kube-system
    resourceVersion: "57"
    selfLink: /api/v1/namespaces/kube-system/configmaps/extension-apiserver-authentication
    uid: 5b6016a4-f7f2-11e8-9183-5254004c4249
- apiVersion: v1
  data:
    config.conf: |-
      apiVersion: kubeproxy.config.k8s.io/v1alpha1
      bindAddress: 0.0.0.0
      clientConnection:
        acceptContentTypes: ""
        burst: 10
        contentType: application/vnd.kubernetes.protobuf
        kubeconfig: /var/lib/kube-proxy/kubeconfig.conf
        qps: 5
      clusterCIDR: 57.144.0.0/12
      configSyncPeriod: 15m0s
      conntrack:
        max: null
        maxPerCore: 32768
        min: 131072
        tcpCloseWaitTimeout: 1h0m0s
        tcpEstablishedTimeout: 24h0m0s
      enableProfiling: false
      healthzBindAddress: 0.0.0.0:10256
      hostnameOverride: ""
      iptables:
        masqueradeAll: false
        masqueradeBit: 14
        minSyncPeriod: 0s
        syncPeriod: 30s
      ipvs:
        excludeCIDRs: null
        minSyncPeriod: 0s
        scheduler: ""
        syncPeriod: 30s
      kind: KubeProxyConfiguration
      metricsBindAddress: 127.0.0.1:10249
      mode: ""
      nodePortAddresses: null
      oomScoreAdj: -999
      portRange: ""
      resourceContainer: /kube-proxy
      udpIdleTimeout: 250ms
    kubeconfig.conf: |-
      apiVersion: v1
      kind: Config
      clusters:
      - cluster:
          certificate-authority: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
          server: https://192.168.80.105:6443
        name: default
      contexts:
      - context:
          cluster: default
          namespace: default
          user: default
        name: default
      current-context: default
      users:
      - name: default
        user:
          tokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
  kind: ConfigMap
  metadata:
    creationTimestamp: "2018-12-04T18:28:16Z"
    labels:
      app: kube-proxy
    name: kube-proxy
    namespace: kube-system
    resourceVersion: "234"
    selfLink: /api/v1/namespaces/kube-system/configmaps/kube-proxy
    uid: 5da5f572-f7f2-11e8-9183-5254004c4249
- apiVersion: v1
  data:
    ClusterConfiguration: |
      apiServer:
        extraArgs:
          authorization-mode: Node,RBAC
        timeoutForControlPlane: 4m0s
      apiVersion: kubeadm.k8s.io/v1beta1
      certificatesDir: /etc/kubernetes/pki
      clusterName: kubernetes
      controlPlaneEndpoint: ""
      controllerManager: {}
      dns:
        type: CoreDNS
      etcd:
        local:
          dataDir: /var/lib/etcd
      imageRepository: k8s.gcr.io
      kind: ClusterConfiguration
      kubernetesVersion: v1.13.0
      networking:
        dnsDomain: sbezverk.cisco.com
        podSubnet: 57.144.0.0/12
        serviceSubnet: 57.160.0.0/12
      scheduler: {}
    ClusterStatus: |
      apiEndpoints:
        kube-5.sbezverk.cisco.com:
          advertiseAddress: 192.168.80.105
          bindPort: 6443
      apiVersion: kubeadm.k8s.io/v1beta1
      kind: ClusterStatus
  kind: ConfigMap
  metadata:
    creationTimestamp: "2018-12-04T18:28:14Z"
    name: kubeadm-config
    namespace: kube-system
    resourceVersion: "179"
    selfLink: /api/v1/namespaces/kube-system/configmaps/kubeadm-config
    uid: 5cffc8a9-f7f2-11e8-9183-5254004c4249
- apiVersion: v1
  data:
    kubelet: |
      address: 0.0.0.0
      apiVersion: kubelet.config.k8s.io/v1beta1
      authentication:
        anonymous:
          enabled: false
        webhook:
          cacheTTL: 2m0s
          enabled: true
        x509:
          clientCAFile: /etc/kubernetes/pki/ca.crt
      authorization:
        mode: Webhook
        webhook:
          cacheAuthorizedTTL: 5m0s
          cacheUnauthorizedTTL: 30s
      cgroupDriver: cgroupfs
      cgroupsPerQOS: true
      clusterDNS:
      - 57.160.0.10
      clusterDomain: sbezverk.cisco.com
      configMapAndSecretChangeDetectionStrategy: Watch
      containerLogMaxFiles: 5
      containerLogMaxSize: 10Mi
      contentType: application/vnd.kubernetes.protobuf
      cpuCFSQuota: true
      cpuCFSQuotaPeriod: 100ms
      cpuManagerPolicy: none
      cpuManagerReconcilePeriod: 10s
      enableControllerAttachDetach: true
      enableDebuggingHandlers: true
      enforceNodeAllocatable:
      - pods
      eventBurst: 10
      eventRecordQPS: 5
      evictionHard:
        imagefs.available: 15%
        memory.available: 100Mi
        nodefs.available: 10%
        nodefs.inodesFree: 5%
      evictionPressureTransitionPeriod: 5m0s
      failSwapOn: true
      fileCheckFrequency: 20s
      hairpinMode: promiscuous-bridge
      healthzBindAddress: 127.0.0.1
      healthzPort: 10248
      httpCheckFrequency: 20s
      imageGCHighThresholdPercent: 85
      imageGCLowThresholdPercent: 80
      imageMinimumGCAge: 2m0s
      iptablesDropBit: 15
      iptablesMasqueradeBit: 14
      kind: KubeletConfiguration
      kubeAPIBurst: 10
      kubeAPIQPS: 5
      makeIPTablesUtilChains: true
      maxOpenFiles: 1000000
      maxPods: 110
      nodeLeaseDurationSeconds: 40
      nodeStatusReportFrequency: 1m0s
      nodeStatusUpdateFrequency: 10s
      oomScoreAdj: -999
      podPidsLimit: -1
      port: 10250
      registryBurst: 10
      registryPullQPS: 5
      resolvConf: /etc/resolv.conf
      rotateCertificates: true
      runtimeRequestTimeout: 2m0s
      serializeImagePulls: true
      staticPodPath: /etc/kubernetes/manifests
      streamingConnectionIdleTimeout: 4h0m0s
      syncFrequency: 1m0s
      volumeStatsAggPeriod: 1m0s
  kind: ConfigMap
  metadata:
    creationTimestamp: "2018-12-04T18:28:14Z"
    name: kubelet-config-1.13
    namespace: kube-system
    resourceVersion: "182"
    selfLink: /api/v1/namespaces/kube-system/configmaps/kubelet-config-1.13
    uid: 5d0153ba-f7f2-11e8-9183-5254004c4249
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
```

etcd log:
```
2018-12-04 19:53:51.156356 I | etcdmain: etcd Version: 3.2.24
2018-12-04 19:53:51.156419 I | etcdmain: Git SHA: 420a45226
2018-12-04 19:53:51.156437 I | etcdmain: Go Version: go1.8.7
2018-12-04 19:53:51.156447 I | etcdmain: Go OS/Arch: linux/amd64
2018-12-04 19:53:51.156452 I | etcdmain: setting maximum number of CPUs to 8, total number of available CPUs is 8
2018-12-04 19:53:51.156530 N | etcdmain: the server is already initialized as member before, starting as etcd member...
2018-12-04 19:53:51.156557 I | embed: peerTLS: cert = /etc/kubernetes/pki/etcd/peer.crt, key = /etc/kubernetes/pki/etcd/peer.key, ca = , trusted-ca = /etc/kubernetes/pki/etcd/ca.crt, client-cert-auth = true
2018-12-04 19:53:51.157437 I | embed: listening for peers on https://192.168.80.105:2380
2018-12-04 19:53:51.157511 I | embed: listening for client requests on 127.0.0.1:2379
2018-12-04 19:53:51.157560 I | embed: listening for client requests on 192.168.80.105:2379
2018-12-04 19:53:51.160173 I | etcdserver: name = kube-5.sbezverk.cisco.com
2018-12-04 19:53:51.160191 I | etcdserver: data dir = /var/lib/etcd
2018-12-04 19:53:51.160209 I | etcdserver: member dir = /var/lib/etcd/member
2018-12-04 19:53:51.160214 I | etcdserver: heartbeat = 100ms
2018-12-04 19:53:51.160230 I | etcdserver: election = 1000ms
2018-12-04 19:53:51.160235 I | etcdserver: snapshot count = 10000
2018-12-04 19:53:51.160243 I | etcdserver: advertise client URLs = https://192.168.80.105:2379
2018-12-04 19:53:51.187860 I | etcdserver: restarting member 30edf8082ba0f88f in cluster 2df7cf38564db9e4 at commit index 4273
2018-12-04 19:53:51.188247 I | raft: 30edf8082ba0f88f became follower at term 30
2018-12-04 19:53:51.188266 I | raft: newRaft 30edf8082ba0f88f [peers: [], term: 30, commit: 4273, applied: 0, lastindex: 4273, lastterm: 30]
2018-12-04 19:53:51.189606 I | mvcc: restore compact to 801
2018-12-04 19:53:51.198934 W | auth: simple token is not cryptographically signed
2018-12-04 19:53:51.201152 I | etcdserver: starting server... [version: 3.2.24, cluster version: to_be_decided]
2018-12-04 19:53:51.201858 I | embed: ClientTLS: cert = /etc/kubernetes/pki/etcd/server.crt, key = /etc/kubernetes/pki/etcd/server.key, ca = , trusted-ca = /etc/kubernetes/pki/etcd/ca.crt, client-cert-auth = true
2018-12-04 19:53:51.202364 I | etcdserver/membership: added member 30edf8082ba0f88f [https://192.168.80.105:2380] to cluster 2df7cf38564db9e4
2018-12-04 19:53:51.202486 N | etcdserver/membership: set the initial cluster version to 3.2
2018-12-04 19:53:51.202548 I | etcdserver/api: enabled capabilities for version 3.2
2018-12-04 19:53:52.188715 I | raft: 30edf8082ba0f88f is starting a new election at term 30
2018-12-04 19:53:52.188748 I | raft: 30edf8082ba0f88f became candidate at term 31
2018-12-04 19:53:52.188768 I | raft: 30edf8082ba0f88f received MsgVoteResp from 30edf8082ba0f88f at term 31
2018-12-04 19:53:52.188781 I | raft: 30edf8082ba0f88f became leader at term 31
2018-12-04 19:53:52.188801 I | raft: raft.node: 30edf8082ba0f88f elected leader 30edf8082ba0f88f at term 31
2018-12-04 19:53:52.189594 I | etcdserver: published {Name:kube-5.sbezverk.cisco.com ClientURLs:[https://192.168.80.105:2379]} to cluster 2df7cf38564db9e4
2018-12-04 19:53:52.189624 I | embed: ready to serve client requests
2018-12-04 19:53:52.189738 I | embed: ready to serve client requests
2018-12-04 19:53:52.189874 I | embed: serving client requests on 127.0.0.1:2379
2018-12-04 19:53:52.189885 I | embed: serving client requests on 192.168.80.105:2379
2018-12-04 19:55:18.458409 N | pkg/osutil: received terminated signal, shutting down...
WARNING: 2018/12/04 19:55:18 grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:2379: getsockopt: connection refused"; Reconnecting to {127.0.0.1:2379 0  <nil>}
WARNING: 2018/12/04 19:55:19 grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:2379: getsockopt: connection refused"; Reconnecting to {127.0.0.1:2379 0  <nil>}
WARNING: 2018/12/04 19:55:21 grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:2379: getsockopt: connection refused"; Reconnecting to {127.0.0.1:2379 0  <nil>}
WARNING: 2018/12/04 19:55:23 grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:2379: getsockopt: connection refused"; Reconnecting to {127.0.0.1:2379 0  <nil>}
2018-12-04 19:55:25.458776 W | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Unavailable desc = body closed by handler")
2018-12-04 19:55:25.458957 W | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Unavailable desc = body closed by handler")
2018-12-04 19:55:25.459046 W | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Unavailable desc = body closed by handler")
2018-12-04 19:55:25.459109 W | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Unavailable desc = body closed by handler")
2018-12-04 19:55:25.459164 W | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Unavailable desc = body closed by handler")
2018-12-04 19:55:25.459242 W | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Unavailable desc = body closed by handler")
2018-12-04 19:55:25.459301 W | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Unavailable desc = body closed by handler")
2018-12-04 19:55:25.459342 W | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Unavailable desc = body closed by handler")
2018-12-04 19:55:25.459373 W | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Unavailable desc = body closed by handler")
2018-12-04 19:55:25.459388 W | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Unavailable desc = body closed by handler")
2018-12-04 19:55:25.459467 W | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Unavailable desc = body closed by handler")
2018-12-04 19:55:25.459532 W | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Unavailable desc = body closed by handler")
2018-12-04 19:55:25.459546 W | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Unavailable desc = body closed by handler")
2018-12-04 19:55:25.459572 W | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Unavailable desc = body closed by handler")
2018-12-04 19:55:25.459589 W | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Unavailable desc = body closed by handler")
2018-12-04 19:55:25.459605 W | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Unavailable desc = body closed by handler")
2018-12-04 19:55:25.459789 W | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Unavailable desc = body closed by handler")
2018-12-04 19:55:25.459809 W | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Unavailable desc = body closed by handler")
2018-12-04 19:55:25.459874 W | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Unavailable desc = body closed by handler")
2018-12-04 19:55:25.460054 W | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Unavailable desc = body closed by handler")
2018-12-04 19:55:25.460200 W | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Unavailable desc = body closed by handler")
2018-12-04 19:55:25.460390 W | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Unavailable desc = body closed by handler")
2018-12-04 19:55:25.460529 W | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Unavailable desc = body closed by handler")
2018-12-04 19:55:25.460549 W | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Unavailable desc = body closed by handler")
2018-12-04 19:55:25.460766 W | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Unavailable desc = body closed by handler")
2018-12-04 19:55:25.460822 W | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Unavailable desc = body closed by handler")
2018-12-04 19:55:25.460912 W | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Unavailable desc = body closed by handler")
2018-12-04 19:55:25.460991 W | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Unavailable desc = body closed by handler")
2018-12-04 19:55:25.461054 W | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Unavailable desc = body closed by handler")
2018-12-04 19:55:25.461068 W | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Unavailable desc = body closed by handler")
2018-12-04 19:55:25.461088 W | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Unavailable desc = body closed by handler")
2018-12-04 19:55:25.461122 W | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Unavailable desc = body closed by handler")
2018-12-04 19:55:25.461199 W | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Unavailable desc = body closed by handler")
2018-12-04 19:55:25.461225 W | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Unavailable desc = body closed by handler")
2018-12-04 19:55:25.461254 W | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Unavailable desc = body closed by handler")
2018-12-04 19:55:25.461362 W | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Unavailable desc = body closed by handler")
2018-12-04 19:55:25.461453 W | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Unavailable desc = body closed by handler")
WARNING: 2018/12/04 19:55:25 grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: Error while dialing dial tcp 192.168.80.105:2379: getsockopt: connection refused"; Reconnecting to {192.168.80.105:2379 0  <nil>}
2018-12-04 19:55:25.461493 W | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Unavailable desc = body closed by handler")
2018-12-04 19:55:25.461522 W | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Unavailable desc = body closed by handler")
2018-12-04 19:55:25.461539 W | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Unavailable desc = body closed by handler")
2018-12-04 19:55:25.461557 W | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Unavailable desc = body closed by handler")
2018-12-04 19:55:25.462158 W | etcdserver/api/v3rpc: failed to receive watch request from gRPC stream ("rpc error: code = Unavailable desc = body closed by handler")
2018-12-04 19:55:25.958973 I | etcdserver: skipped leadership transfer for single member cluster
WARNING: 2018/12/04 19:55:25 grpc: addrConn.transportMonitor exits due to: context canceled
WARNING: 2018/12/04 19:55:25 grpc: addrConn.transportMonitor exits due to: context canceled
```

**Comment by mauilion** [comment]
From the logs you can't connect to 127.0.0.1:2379 This seems to be the main issue. As if that fails you will also fail healthcheck and the pod will be restarted (flap)

**Comment by sbezverk** [comment]
@mauilion I agree but what is not clear why after restart, I can curl etcd with command you suggested. So it seem the issue with with health check and not the actual connectivity.

**Comment by neolit123** [comment]
so, if you are running CentOS 7.x with Docker 1.13.1.87 and seeing messages like:
```
"process_linux.go:110: decoding init error from pipe caused \"read parent: connection reset by peer
```

this is an issue with Docker and you should either downgrade to 1.13.1-75 or just install 18.06 which is the recommended docker version in k8s 1.12 and 1.13.

i will add a troubleshooting guide entry about that.

**Comment by mauilion** [comment]
yep can confirm. I was able to reproduce this with:
vagrant init generic/centos7
yum update 
then running through the docs for kubeadm init on 1.13.

I see that the kubelet is killing etcd after a bit as it can't exec into the etcd container. 

I verified that also can not as a root user exec into a running container. 

I then downgraded per the ticket that neolit23 found and the issue went away. 

I've also started an generic/ubuntu1604 vagrant and was unable to reproduce this there.
seems like there is something specific about the .87 patch that breaks stuff for centos and the like.
